src/marqo/s2_inference/clip_utils.py

```diff
@@ -475,134 +483,235 @@ def encode(self, inputs: Union[str, ImageType, List[Union[str, ImageType]]],
-                              f"Please check you cuda availability or try the fp32 version `{self.model_type.replace('fp16/','')}`"
-                              f"Check `https://docs.marqo.ai/0.0.13/Models-Reference/dense_retrieval/#generic-clip-models` for more info.")
+                           f"FP16 clip model `{self.model_type}` is only available with device `cuda`.\n"
+                           f"With current device `{self.device}`, the model will be loaded in `float32` mode. \n"
+                           f"Please check you cuda availability or try the fp32 version `{self.model_type.replace('fp16/', '')}`"
+                           f"Check `https://docs.marqo.ai/0.0.13/Models-Reference/dense_retrieval/#generic-clip-models` for more info.")
 
         self.model_name = self.model_type.replace("fp16/", "")
 
-
     def load(self) -> None:
         # https://github.com/openai/CLIP/issues/30
-        self.model, self.preprocess = clip.load(self.model_name, device=self.device, jit=False, download_root=ModelCache.clip_cache_path)
+        self.model, self.preprocess = clip.load(self.model_name, device=self.device, jit=False,
+                                                download_root=ModelCache.clip_cache_path)
         self.model = self.model.to(self.device)
         self.tokenizer = clip.tokenize
         self.model.eval()
 
 
-class OPEN_CLIP(CLIP):
-    def __init__(self, model_type: str = "open_clip/ViT-B-32-quickgelu/laion400m_e32", device: str = None,  embedding_dim: int = None,
-                            truncate: bool = True, **kwargs) -> None:
-        super().__init__(model_type, device,  embedding_dim, truncate , **kwargs)
+class OPEN_CLIP(AbstractCLIPModel):
+    def __init__(
+            self,
+            model_type: str = "open_clip/ViT-B-32-quickgelu/laion400m_e32",
+            device: Optional[str] = None,
+            embedding_dim: Optional[int] = None,
+            truncate: bool = True,
+            model_properties: Optional[Dict] = None,
+            model_auth: Optional[Dict] = None
+    ) -> None:
+
+        super().__init__(model_type, device, embedding_dim, truncate, model_properties)
+
+        # model_auth gets passed through add_docs and search requests:
+        self.model_auth = model_auth
+        self.preprocess_config = None
         self.model_name = model_type.split("/", 3)[1] if model_type.startswith("open_clip/") else model_type
         self.pretrained = model_type.split("/", 3)[2] if model_type.startswith("open_clip/") else model_type
+        self.model_properties = OpenCLIPModelProperties(**self.model_properties)
 
     def load(self) -> None:
-        # https://github.com/mlfoundations/open_clip
-        path = self.model_properties.get("localpath", None) or self.model_properties.get("url", None)
+        """Load the open_clip model and tokenizer."""
+        if self.model_properties.url or self.model_properties.model_location:
```

Comment:
If you mean to check for None, be explicit

Reply:
done